src/Microsoft.AspNetCore.Sockets.Client.Http/Internal/ConnectionLogScope.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace Microsoft.AspNetCore.Sockets.Http.Internal
+{
+    public class ConnectionLogScope : IReadOnlyList<KeyValuePair<string, object>>
+    {
+        private string _cachedToString;
+        private string _connectionId;
+
+        public string ConnectionId
+        {
+            get
+            {
+                return _connectionId;
+            }
+            set
+            {
+                _cachedToString = null;
+                _connectionId = value;
+            }
+        }
+
+        public KeyValuePair<string, object> this[int index]
```

Comment:
A little confused by this

Reply:
?

Reply:
I guess the question is why is it not returning an `Object` but a KeyValuePair...

Reply:
Resolved in person!